xcdat/averager.py

```diff
@@ -20,275 +23,311 @@
-    """
-    # ensure iterable array
-    region_lon_bounds = np.asarray(region_lon_bounds)
-    region_lon_bounds = np.atleast_1d(region_lon_bounds)
-    # get the max / min bounds for each grid cell
-    min_grid_lon_bounds = np.min(grid_lon_bounds, axis=1)
-    max_grid_lon_bounds = np.max(grid_lon_bounds, axis=1)
-    # iterate over regional bounds and check to see if they fall within
-    # any grid cell bounds.
-    in_bounds = True
-    for lon in region_lon_bounds:
-        inds = np.where((min_grid_lon_bounds <= lon) &
-                        (max_grid_lon_bounds >= lon))[0]
-        if len(inds) == 0:
-            in_bounds = False
-    return in_bounds
-
-
-def _switch_lon_bounds(lon, totype='to360'):
-    """Switch longitude axis orientation either 'to360' (0 to 360) or 'to180'
-    (0 to 180).
+
+def _swap_region_lon_axis(
+    region_lon: RegionAxisBounds, to: Literal["180", "360"]
+) -> RegionAxisBounds:
+    """Swap region longitude axis with "180" (-180 to 180) or "360" (0 to 360).
 
     Parameters
     ----------
-    lon : array
-        array of longitude values to convert
-    totype : str, optional
-        string specifying conversion type ('to180' or 'to360')
+    region_lon : RegionAxisBounds
+        List of longitude values to convert.
+    to : Literal["180", "360"]
+        Specify conversion type.
 
     Returns
     -------
-    array
-        Array of converted longitude values
+    RegionAxisBounds
+        List of converted longitude values.
 
     Notes
     -----
     This does not re-order the values in any way; it only converts the values
-    in-place between longitude conventions (0 to 360 or -180 to 180).
+    in-place between longitude conventions (-180 to 180) or (0 to 360).
+    """
+    rl_nparray = np.array(region_lon)
+
+    if to == "360":
+        inds = np.where(rl_nparray < 0)[0]
+        rl_nparray[inds] = rl_nparray[inds] + 360
+    elif to == "180":
+        inds = np.where(rl_nparray > 180)[0]
+        rl_nparray[inds] = rl_nparray[inds] - 360
+
+    return rl_nparray.tolist()
+
+
+def _is_region_lon_in_domain(
+    region_bounds: RegionAxisBounds, domain_bounds: xr.DataArray
+) -> bool:
+    """Check if a region longitude bounds fall within domain longitude bounds.
+
+    Parameters
+    ----------
+    region_bounds : RegionAxisBounds
+        Region longitude bounds.
+    domain_bounds : xr.DataArray
+        Domain longitude bounds.
+
+    Returns
+    -------
+    bool
+        True if all region longitude bounds are within grid longitude bounds.
+    """
+    min_domain_bounds = domain_bounds.min(axis=1)
+    max_domain_bounds = domain_bounds.max(axis=1)
+
+    in_domain = True
+    for bound in region_bounds:
+        count_in_domain = np.where(
+            (min_domain_bounds <= bound) & (max_domain_bounds >= bound)
+        )[0]
+        if count_in_domain.size == 0:
+            in_domain = False
+
+    return in_domain
+
+
+def _get_axis_weights(
```

Comment:
This function has been refactored into separate smaller functions `_get_area_weights()` and `_get_uniform_weights()` + `_scale_dim_to_region()`.